app/models/parent_object.rb

```diff
@@ -194,41 +195,32 @@ def upsert_preservica_ingest_child_objects(preservica_ingest_hash)
     PreservicaIngest.insert_all(preservica_ingest_hash)
   end
 
-  # rubocop:disable Metrics/MethodLength
-  def sync_from_preservica
-    # get list of child preservica_content_object_uri currently associated with parent
-    local_content_uris = []
-    child_objects.map do |co|
-      local_content_uris << co.preservica_content_object_uri
+  def sync_from_preservica(local_children_hash, preservica_children_hash)
+    # iterate through local hashes and remove any children no longer found on preservica
+    local_children_hash.each_value do |value|
+      co = ChildObject.find_by(order: value[:order])
+      co.destroy unless found_in_preservica(value[:content_uri], preservica_children_hash)
     end
-    # get list of content object uris from preservica
-    preservica_content_uris = []
-    PreservicaImageService.new(preservica_uri, admin_set.key).image_list(preservica_representation_name).map do |co|
-      preservica_content_uris << co[:preservica_content_object_uri]
-    end
-    local_content_uris.map do |local|
-      # keep child objects that have a match in preservica
-      next if preservica_content_uris.include?(local)
-
-      # delete child objects that don't have a match in preservica
-      co = ChildObject.find_by(preservica_content_object_uri: local)
-      co.destroy
+    # iterate through preservica and update when local version found
+    preservica_children_hash.each_value do |value|
+      co = ChildObject.find_by(preservica_content_object_uri: value[:content_uri])
+      next if co.nil?
+      co.order = value[:order]
+      co.preservica_content_object_uri = value[:content_uri]
+      co.preservica_generation_uri = value[:generation_uri]
+      co.preservica_bitstream_uri = value[:bitstream_uri]
+      co.save
     end
+    # create child records for any new items in preservica
+    create_child_records
```

Comment:
Is there an issue with calling `create_child_records`, which calls `array_of_child_hashes_from_preservica` ?  The comment says that the latter method downloads all of the child images from Preservica, but we should only need to download the new ones during a sync.

Reply:
There is a scrub of already existing child images [here](https://github.com/yalelibrary/yul-dc-management/pull/992/files#diff-acc421bf5477cf1667d0b61e9ded247f3b42335a2425020920d8c20919ff0384R155) in `validate_child_hashes` that removes any child hashes that are duplicates and this should ensure only new child images are added to the parent.